MovieCave - MVVM/Movie-Cave/Movie-Cave/View/AddMovieView/AddMovieViewController.swift

```diff
@@ -0,0 +1,209 @@
+    private let starRaitingView = StartsRaitingView()
+    private var cancellables: [AnyCancellable] = []
+
+    //MARK: - LifeCycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setUpDelegates()
+        raitingView.addSubview(starRaitingView)
+        starRaitingView.frame = raitingView.bounds
+        addGenreChecks()
+        gerneViewSetUp()
+        setUpDoneButton()
+        setUpBinders()
+    }
+    
+    //MARK: - IBActions
+    @IBAction func imageButtonTapped(_ sender: UIButton) {
+        imagePicker.sourceType = .photoLibrary
+        present(imagePicker, animated: true, completion: nil)
+    }
+    
+    
+    @IBAction func doneButtonTapped(_ sender: UIButton) {
+        viewModel.addFinishedMovie()
+        navigationController?.popViewController(animated: true)
+    }
+    
+}
+
+//MARK: - CustomCheckBoxViewDelegate
+extension AddMovieViewController: CustomCheckBoxViewDelegate {
+    func checkedBox(text: String) {
+        viewModel.checkBox(checkBoxText: text)
+    }
+
+}
+
+//MARK: - StarRaitingDelegate
+extension AddMovieViewController: StarsRaitingViewDelegate {
+    
+    func starsReceived(rate: Int) {
+        viewModel.starsRatingReceived(rating: rate)
+    }
+    
+}
+
+//MARK: - TextFiled Delegate
+extension AddMovieViewController: UITextFieldDelegate {
+
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        guard let movieTitle = movieTitleTextField.text,
+              let movieDescribtion = describtionTextField.text else {
+            return
+        }
+        
+        if textField == movieTitleTextField {
+            viewModel.receiveMovieTitle(title: movieTitle)
+        } else if textField == describtionTextField {
+            viewModel.receiveMovieDescrption(description: movieDescribtion)
+        }
+
+    }
+
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        movieTitleTextField.resignFirstResponder()
+        describtionTextField.resignFirstResponder()
+      return true
+    }
+    
+}
+
+//MARK: - UIImagePickerControllerDelegate and UINavigationControllerDelegate
+extension AddMovieViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate{
+    
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        guard let image = info[.originalImage] as? UIImage else { return }
+        
+        let imageTitle = UUID().uuidString
+        movieImageButton.setImage(image, for: .normal)
+        image.saveImageToDocumentsDirectory(image: image, withName: imageTitle)
+        viewModel.uploadMovieImage(imageTitle: imageTitle)
+        
+      dismiss(animated: true, completion: nil)
+    }
+
+    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
+      dismiss(animated: true, completion: nil)
+    }
+    
+}
+
+//MARK: - AddMovieViewController UI setUp
+extension AddMovieViewController {
```

Comment:
It would be better to export this extension on separate file, but keeping the binders inside this file